src/extension.ts

```diff
@@ -885,3 +887,33 @@ function isSupportedLanguage(languageId: TextDocument["languageId"]): boolean {
       return false;
   }
 }
+
+// NOTE(gabro): we would normally use the `configurationDefaults` contribution point in the
+// extension manifest but that's currently limited to language-scoped settings in VSCode.
+// We use this method to set global configuration settings such as `files.watcherExclude`
+// and `search.exclude`.
+function configureGlobalSettingsDefaults() {
```

Comment:
```suggestion
function configureWorkspaceSettingsDefaults() {
```
?

Reply:
These are actually global, i.e. they're scoped to the "User" configuration, since they're fairly general.

Reply:
Ach, maybe it would be safer to scope it to user? Especially if we include build tool specific excludes?

Reply:
it is scoped to "user" right now.

To recap, the scopes are:

- Global (i.e. User in the UI)
- Workspace
- WorkspaceFolder

Also, some settings can be scoped to a specific language (but these specific settings cannot).

Since these are very general, I would lean towards scoping them to Global. My idea is that it makes sense to always ignore `.bloop` / `.metals` / `target` and the like in file watching and search results.

Actually, now that I remembered, it's useless to add `search.exclude` since there's an option in VSCode to automatically exclude files ignored by git.

Reply:
Ok, removed now.

Bottom line: as long as we're configure VSCode to exclude very specific files from the file watching such as `.bloop` and `.metals` I think we can safely do that globally.

Reply:
Ok, I think those 3 plus `.ammonite` should be safe to include. The mill's out is usually not that big of a problem and `out` might be too general.

Reply:
I agree 👍 

Are you sure `.ammonite` is used at the project level? I've tried googling and I could only find references for a `~/.ammonite` dir

Reply:
It's only there if you start a BSP script connection.

Reply:
👍 added!